packages/contracts-bedrock/src/L1/SystemConfigInterop.sol

```diff
@@ -48,19 +94,32 @@ contract SystemConfigInterop is SystemConfig {
         }
     }
 
-    /// @notice Adds a chain to the interop dependency set. Can only be called by the owner.
+    /// @notice Adds a chain to the interop dependency set. Can only be called by the dependency manager.
     /// @param _chainId Chain ID of chain to add.
-    function addDependency(uint256 _chainId) external onlyOwner {
+    function addDependency(uint256 _chainId) external {
+        require(
+            msg.sender == Storage.getAddress(DEPENDENCY_MANAGER_SLOT),
+            "SystemConfig: caller is not the dependency manager"
+        );
         OptimismPortal(payable(optimismPortal())).setConfig(
             ConfigType.ADD_DEPENDENCY, StaticConfig.encodeAddDependency(_chainId)
         );
     }
 
-    /// @notice Removes a chain from the interop dependency set. Can only be called by the owner.
+    /// @notice Removes a chain from the interop dependency set. Can only be called by the dependency manager
     /// @param _chainId Chain ID of the chain to remove.
-    function removeDependency(uint256 _chainId) external onlyOwner {
+    function removeDependency(uint256 _chainId) external {
+        require(
+            msg.sender == Storage.getAddress(DEPENDENCY_MANAGER_SLOT),
```

Comment:
Curious if you have considered calling `dependencyManager()` here rather than the lower level API? its a matter of preference, curious if you have an opinion between which is nicer

Reply:
did it so that `dependencyManager()` could be external but i think that since it (`Storage.getAddress(DEPENDENCY_MANAGER_SLOT`) is used twice, using the function would be cleaner (also considering the non-function option saves only a bit of gas).